onset_maproom/calc.py

```diff
@@ -102,50 +99,47 @@ def onset_date(
     dry_spell_search,
     time_coord="T",
 ):
-    """ Calculate onset date.
+    """Calculate onset date.
 
-        Finds the first wet spell of 'wet_spell_length' days where: 
-            Cumulative rain exceeds 'wet_spell_thresh',
-            With at least 'min_wet_days' count of wet days (greater than 'wet_thresh'),
-            Not followed by a dry spell of 'dry_spell_length' days of dry days (not wet),
-            For the following 'dry_spell_search' days
+    Find the first wet spell of `wet_spell_length` days where: 
+        Cumulative rain exceeds `wet_spell_thresh`,
+        With at least `min_wet_days` count of wet days (greater than `wet_thresh`),
+        Not followed by a dry spell of `dry_spell_length` days of dry days (not wet),
+        For the following `dry_spell_search` days
     
-        Parameters
-        ----------
-        daily_rain : DataArray
-                    Array of daily rainfall values.
-        wet_thresh : int
-                    Rainfall threshold to determine wet day.
-        wet_spell_length : int
-                    Length in days of running window when 'wet_thresh' is to be met to define a wet spell.
-        wet_spell_thresh : int
-                    Threshold of rainfall to be reached during 'wet_spell_length'
-                    window to define a wet spell.
-        min_wet_days : int
-                    Minimum number of wet days in 'wet_spell_length' window when it rained at or above
-                    'wet_spell_thresh' to be considered a wet spell.
-        dry_spell_length : int
-                    Length in days of dry spell that would invalidate a wet spell as onset date
-                    if found within the 'dry_spell_search' days following the 'wet_spell_length'-day window
-                    that met the wet spell criteria.
-        dry_spell_search : int
-                    Length in days to search for a 'dry_spell_length'-day dry spell after a wet spell
-                    is found that would invalidate the wet spell as onset date.
-        time_coord : str, optional
-                    Time grid in 'soil_moisture' (the default is "T", which implies
-                    naming convention of time grid as "T" in 'soil_moisture')         
-        Returns
-        -------
-        onset_delta : DataArray    
-                    The time delta rom the first day of daily_rain
-                    to the first wet day in that wet spell.
-        See Also
-        --------
-        Notes
-        -----
-        Examples
-        --------
-
+    Parameters
+    ----------
+    daily_rain : DataArray
+        Array of daily rainfall values.
+    wet_thresh : int
```

Comment:
constant rather than int (can be other than int, e.g. 4.5). Or I am not sure what python/xarray terminology is. But int is a data type whereas constant is an object type. It looks like when it's a constant that must be of one type, it's ok to "simplify" by saying just int. But in that case it's not working since it can be a real number.

Reply:
so a float then?